lib/Serialization/Serialization.cpp

```diff
@@ -2369,24 +2369,16 @@ void Serializer::writeDeclAttribute(const DeclAttribute *DA) {
       vjpRef = addDeclRef(attr->getVJPFunction());
     }
 
-    SmallVector<uint32_t, 4> parameters;
-    for (auto param : attr->getParameters()) {
-      switch (param.getKind()) {
-      // The self parameter is uniquely identified by 0x01.
-      case AutoDiffParameter::Kind::Self:
-        parameters.push_back(1);
-        break;
-      // Index parameters are left-shifted by 1.
-      case AutoDiffParameter::Kind::Index:
-        parameters.push_back(param.getIndex() << 1);
-        break;
-      }
-    }
+    auto paramIndices = attr->getParameterIndices();
+    assert(paramIndices && "Checked parameter indices must be resolved");
+    SmallVector<bool, 4> indices;
+    for (unsigned i = 0; i < paramIndices->parameters.size(); i++)
```

Comment:
```suggestion
    for (unsigned i : indices(paramIndices->parameters.size()))
```

Reply:
Thanks! Done in https://github.com/apple/swift/pull/21837/commits/5d8016b1850d755062fba2d9f19c12495569b5ac.